src/main/kotlin/org/neo4j/graphql/handler/filter/OptimizedFilterHandler.kt

```diff
@@ -0,0 +1,369 @@
+                                exposesWith,
+                                levelPassThroughWiths + additionalWiths + listOf(total, count)
+                        )
+                        val where = whereClause(withWithoutWhere, Cypher.name(totalVar), Cypher.name(countVar))
+                        withClauseWithOptionalDistinct(where, levelPassThroughWiths + additionalWiths)
+                    }
+                }
+
+                when (op) {
+                    RelationOperator.SOME -> query = nestingLevelHandler.parseFilter()
+                    RelationOperator.EVERY -> query = nestingLevelHandler.parseFilter(totalAndCountFilter({ withWithoutWhere, total, count ->
+                        withWithoutWhere.where(total.isEqualTo(count))
+                    }))
+                    RelationOperator.SINGLE -> query = nestingLevelHandler.parseFilter(totalAndCountFilter({ withWithoutWhere, total, count ->
+                        withWithoutWhere
+                            .where(total.isEqualTo(count))
+                            .and(total.isEqualTo(Cypher.literalOf(1)))
+                    }))
+                    RelationOperator.NONE -> {
+                        val patternWithoutWhere = Cypher.listBasedOn(createRelation(rel, current as? Node
+                                ?: throw OptimizedQueryException(), relVariable))
+                        if (nestedParsedQuery.and.isNullOrEmpty() && nestedParsedQuery.or.isNullOrEmpty() && nestedParsedQuery.relationPredicates.isNullOrEmpty()) {
+                            if (nestedParsedQuery.fieldPredicates.isNotEmpty()) {
+                                val patternWithoutReturn = addConditions<PatternComprehension.OngoingDefinitionWithoutReturn>(
+                                        relVariable, relVariableName, nestedParsedQuery.fieldPredicates, filterParams) { where, condition ->
+                                    where?.and(condition) ?: patternWithoutWhere.where(condition)
+                                } ?: patternWithoutWhere
+                                val where = query
+                                    .where(Functions.size(patternWithoutReturn.returning(Cypher.literalTrue())).isEqualTo(Cypher.literalOf(0)))
+                                query = withClauseWithOptionalDistinct(where, levelPassThroughWiths)
+                            }
+                        } else {
+                            throw OptimizedQueryException()
+                        }
+                    }
+                    else -> throw IllegalStateException("$op should not be set for relationPredicates")
+                }
+            }
+            return query
+        }
+
+        private fun handleCombinations(passedQuery: StatementBuilder.OrderableOngoingReadingAndWithWithoutWhere): StatementBuilder.OrderableOngoingReadingAndWithWithoutWhere {
+            if ( parsedQuery.or.isNullOrEmpty() && parsedQuery.and.isNullOrEmpty()){
```

Comment:
I think we had that predicate already before

Reply:
Nope